components/browser/state/src/main/java/mozilla/components/browser/state/state/BrowserState.kt

```diff
@@ -31,6 +31,7 @@ data class BrowserState(
     val containers: Map<String, ContainerState> = emptyMap(),
     val extensions: Map<String, WebExtensionState> = emptyMap(),
     val media: MediaState = MediaState(),
+    val mediaSession: MediaSessionState = MediaSessionState(),
```

Comment:
My biggest question here is probably whether we should move this to `SessionState` and give every `SessionState` an optional `MediaSessionState?` instead of having one global `MediaSessionState`. Some advantage I see:
* We are independent from event ordering issues. If GeckoView will activate/deactivate sessions in an unexpected order then we may end up keeping the deactive state here and override the activate state.
* It's closer to the state in GV and we can support multiple sessions in the future.
* Getting the global state is not hard: We can get it from the list of tabs. And if that is too expensive then we can have a middleware that will aggregate a state for us (like we do for media right now, just a bit simpler).
* I feel like this could also make the `EngineSession.Observer` / `EngineObserver` / `BrowserStore` flow simpler since you do not need to pipe all the things through it every time (like everyone of them gets the whole session object and you need to write multiple values from it to make sure you got everything) and instead can only pass the changed data along (e.g. only the metadata).

What do you think?

Reply:
Understood.  I can see the advantages to move this into the `SessionState`.  I'll update this PR.  Thanks!